app/src/main/kotlin/com/wire/android/util/ImageUtil.kt

```diff
@@ -0,0 +1,76 @@
+package com.wire.android.util
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.Matrix
+import android.net.Uri
+import androidx.exifinterface.media.ExifInterface
+import java.io.ByteArrayOutputStream
+
+/**
+ * Rotates the image to a [ExifInterface.ORIENTATION_NORMAL] in case it's rotated with a different orientation
+ *
+ * @param exif Exif interface for of the image to rotate
+ * @return Bitmap the rotated bitmap or the same in case there is no rotation performed
+ */
+private fun Bitmap.rotateImageToNormalOrientation(exif: ExifInterface?): Bitmap {
+    val orientation = exif?.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
+    val matrix = Matrix()
+    when (orientation) {
+        ExifInterface.ORIENTATION_NORMAL -> return this
+        ExifInterface.ORIENTATION_ROTATE_180 -> matrix.postRotate(180f)
+        ExifInterface.ORIENTATION_ROTATE_90 -> matrix.postRotate(90f)
+        ExifInterface.ORIENTATION_ROTATE_270 -> matrix.postRotate(270f)
+        else -> return this
+    }
+
+    return try {
+        val rotated = Bitmap.createBitmap(this, 0, 0, this.width, this.height, matrix, true)
+        this.recycle()
+        rotated
+    } catch (exception: Exception) {
+        this
+    }
+}
+
+/**
+ * Given an image [uri] rotates the image to a [ExifInterface.ORIENTATION_NORMAL] and overwrite the original un-rotated image
+ * Also compress the data to have an efficient data management, without losing quality
+ *
+ * In case of failure, just use the same picture, do nothing
+ *
+ * @param uri the image location on which the operation will be performed
+ * @param context
+ */
+fun postProcessCapturedAvatar(uri: Uri, context: Context) {
+    try {
+        val rawImage = uri.toByteArray(context)
+        val avatarBitmap = rawImage.toBitmap() ?: return
+
+        // Rotate if needed
+        val exifInterface = context.contentResolver.openInputStream(uri).use { inputStream ->
+            return@use inputStream?.let { ExifInterface(it) }
+        }
+        val normalizedAvatar = avatarBitmap.rotateImageToNormalOrientation(exifInterface)
+
+        // Compress image
+        val rawCompressedImage = compressImage(normalizedAvatar)
+
+        // Save to fixed path
+        rawCompressedImage?.let { getWritableTempAvatarUri(it, context) }
+    } catch (exception: Exception) {
+        // NOOP: None post process op performed
+    }
+}
+
+/**
+ * Compress image to save some disk space and memory
+ */
+private fun compressImage(imageBitmap: Bitmap): ByteArray? {
+    val byteArrayOutputStream = ByteArrayOutputStream()
+    imageBitmap.compress(Bitmap.CompressFormat.JPEG, 90, byteArrayOutputStream)
```

Comment:
Do we want to compress it even more, say 50/60% and extract this to a variable that can be reused when sharing media files later?